mmdet/models/dense_heads/decoupled_solo_head.py

```diff
@@ -0,0 +1,505 @@
+import numpy as np
+import mmcv
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import ConvModule, bias_init_with_prob, normal_init
+from mmdet.core import multi_apply, bbox2roi, matrix_nms
+from ..builder import HEADS, build_loss
+from .base_dense_seg_head import BaseDenseSegHead
+
+INF = 1e8
+
+from scipy import ndimage
+
+def points_nms(heat, kernel=2):
+    # kernel must be 2
+    hmax = nn.functional.max_pool2d(
+        heat, (kernel, kernel), stride=1, padding=1)
+    keep = (hmax[:, :, :-1, :-1] == heat).float()
+    return heat * keep
+
+def dice_loss(input, target):
+    input = input.contiguous().view(input.size()[0], -1)
+    target = target.contiguous().view(target.size()[0], -1).float()
+
+    a = torch.sum(input * target, 1)
+    b = torch.sum(input * input, 1) + 0.001
+    c = torch.sum(target * target, 1) + 0.001
+    d = (2 * a) / (b + c)
+    return 1-d
+
+@HEADS.register_module()
+class DecoupledSOLOHead(BaseDenseSegHead):
+    """SOLO: Segmenting Objects by Locations 
+    https://arxiv.org/abs/1912.04488 
+    """ 
+
+    def __init__(self,
+                 num_classes,
+                 in_channels,
+                 seg_feat_channels=256,
+                 stacked_convs=4,
+                 strides=(4, 8, 16, 32, 64),
+                 base_edge_list=(16, 32, 64, 128, 256),
+                 scale_ranges=((8, 32), (16, 64), (32, 128), (64, 256), (128, 512)),
+                 sigma=0.4,
+                 num_grids=None,
+                 cate_down_pos=0,
+                 background_label=None,
+                 loss_ins=None,
```

Comment:
To unify the name of losses with other heads. It might be better to use `loss_mask` and `loss_cls`.